src/nats/mod.rs

```diff
@@ -0,0 +1,97 @@
+use testcontainers::{core::WaitFor, Image, ImageArgs};
+
+const NAME: &str = "nats";
+const TAG: &str = "2.10.14";
+
+/// Nats image for [testcontainers](https://crates.io/crates/testcontainers).
+///
+/// This image is based on the official [Nats](https://hub.docker.com/_/nats) image.
+#[derive(Debug, Default)]
+pub struct Nats {
+    _private: (),
+}
+
+#[derive(Debug, Clone)]
+pub struct NatsServerArgs {
+    pub user: Option<String>,
+    pub pass: Option<String>,
```

Comment:
I think the last request:

Let's keep them private and expose builder methods: `with_user` and `with_pass`

It also will help to avoid breaking changes

Reply:
https://github.com/testcontainers/testcontainers-rs-modules-community/pull/123/commits/5042849ea41a48c687b4e0d089ebe84e435457ae

Reply:
https://github.com/testcontainers/testcontainers-rs-modules-community/pull/123/commits/5042849ea41a48c687b4e0d089ebe84e435457ae